### ISSUE TITLE
[HOLD for payment 2024-10-01] [$250] [Android] When App is opened from notification the top safe area is missing

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#48190]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.25-11
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @shubham1206agra 
**Slack conversation:**

## Action Performed:
Break down in numbered steps

1. [User A] Send a message to user B with chat history
1. [User B] Open the notification received.

## Expected Result:
Describe what you think should've happened

There is a safe padding at the top of the app for status bar

## Actual Result:
Describe what actually happened

There is no safe padding

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

You have to close and reopen the app again

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![image](https://github.com/user-attachments/assets/a624c48a-85ac-42ce-bb7a-f3a4f4c55aca)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015d43274e3d80ec35</li>
        <li>Upwork Job ID: 1828856156434772354</li>
        <li>Last Price Increase: 2024-09-18</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @puneetlath</details>

**Comment by francoisl** [comment]
I'm having a hard time running the app in an Android virtual device at the moment, going to open externally to see if anyone can figure it out.

**Comment by francoisl** [comment]
I'm now able to build the app locally in dev, but running into issues getting system notifications to show up at all.

One interesting thing I noticed with the app on my physical device is that I randomly opened it (manually, without tapping a notification), and the top padding was also missing until I killed and reopened the app. 

<img src="https://github.com/user-attachments/assets/56b4d8bd-01d2-4196-92ad-000cecb7e4c3" width="300" />
<img src="https://github.com/user-attachments/assets/14644afd-0eaf-4f90-b4c1-cd98b0aa8103" width="300" />

I can't reproduce anymore after closing and reopening the app multiple times though, so I feel like there might be other conditions for this to happen.

**Comment by francoisl** [comment]
Going to demote as discussed in https://expensify.slack.com/archives/C01GTK53T8Q/p1724857609487429, since we can't consistently reproduce.

**Comment by WoLewicki** [comment]
> I'm now able to build the app locally in dev, but running into issues getting system notifications to show up at all.
One interesting thing I noticed with the app on my physical device is that I randomly opened it (manually, without tapping a notification), and the top padding was also missing until I killed and reopened the app.

@francoisl could you tell if this happened on `dev` or only on `release`? I was not able to reproduce it on `dev`.

**Comment by francoisl** [comment]
On my physical device? It was a staging release version, at the time 9.0.25-1 (the latest staging version was greater than that, but I guess my phone had not auto-updated the app yet)

**Comment by jayeshmangwani** [comment]
@shubham1206agra since you originally reported this issue, I just want to confirm if it’s still happening for you on the latest version.

**Comment by puneetlath** [comment]
It seems to still be happening in some scenarios: https://github.com/Expensify/App/issues/48477. It doesn't seem to necessarily only have to do with notifications as @francoisl experienced it as well and it wasn't from a notification.

**Comment by jayeshmangwani** [comment]
>  It doesn't seem to necessarily only have to do with notifications

I haven't been able to reproduce this issue yet, even after trying with two Android devices (Samsung and Realme). Let's see if anyone can provide steps to consistently reproduce the issue.

**Comment by puneetlath** [comment]
@shubham1206agra are there any specific steps to repro? Also, what device?

**Comment by puneetlath** [comment]
@perunt has said he'll help look into this one.

**Comment by jayeshmangwani** [comment]
Not overdue; the latest update is [here](https://github.com/Expensify/App/issues/48190#issuecomment-2344071080)

**Comment by jayeshmangwani** [comment]
I was also able to reproduce this issue using the description steps, but for some reason, I am not receiving notifications on my Android devices today

**Comment by puneetlath** [comment]
@perunt any luck so far?

**Comment by perunt** [comment]
Potentially, I found the issue, but to say for sure, I need to be able to receive notifications. And for some reason, I stopped receiving them

**Comment by jayeshmangwani** [comment]
@perunt If you're not receiving notifications, this thread https://expensify.slack.com/archives/C01GTK53T8Q/p1725887904325139?thread_ts=1725353642.370149&cid=C01GTK53T8Q might help you.

**Comment by perunt** [comment]
Thanks a lot!
I started to receive notifications alert in my console in OldDot
<img width="267" alt="image" src="https://github.com/user-attachments/assets/c34caf9b-2d01-48c0-852a-11e439acf550">
but not on my phone. I will investigate it further

**Comment by wildan-m** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The top safe area is missing when opening the app from a notification on Android.

### What is the root cause of that problem?

The return value of `StatusBar.currentHeight` is sometimes 0 when called from a notification. This issue may be related to the initialization timing of `currentHeight` in React Native. Even after waiting for loading to finish, `StatusBar.currentHeight` still remains 0.

### What changes do you think we should make in order to solve the problem?
Consider using `insetsWithDefault.top` as a quick fix if `StatusBar.currentHeight` is zero.

src/components/SafeAreaConsumer/index.android.tsx
```tsx
                const androidInsets = {
                    ...insetsWithDefault,
                    top: StatusBar.currentHeight || insetsWithDefault.top,
                };
```

Or 

```tsx
                const androidInsets = {
                    ...insetsWithDefault,
                    top: StatusBar.currentHeight ? StatusBar.currentHeight : insetsWithDefault.top,
                };
```

### What alternative solutions did you explore? (Optional)
N/A

**Comment by perunt** [comment]
I'd like to clarify the issue a bit. It's not just occurring during the initialization phase, as we would have received the correct status bar height after switching screens if that were the case. Instead, the status bar height remains 0 throughout the app's lifecycle, indicating that the StatusBar functionality itself is compromised.

To address this, we need a two-step approach:
1. Implement a JavaScript workaround as a temporary fix. This might cause some visual jumpiness, but it's necessary for now. 
I've reached out to the original author of the code for more details on this.
2. Tackle the root cause of the problem. It's highly likely that this issue stems from react-native-keyboard-controller, as it replaces the default StatusBar implementation.

I'm digging into it right now, but not being able to send a notification has slowed me down a bit 🥲

**Comment by jayeshmangwani** [comment]
> not being able to send a notification has slowed me down a bit 🥲|

That's strange. Pasting the command `NVP.set('tryNewDot', {classicRedirect: {dismissed: false}});` on OD command line console for the receiver doesn't solve the notification issue. I faced the same issue, but after pasting the above command, it started sending me new notifications.

**Comment by perunt** [comment]
Is it enabled by default for newly created accounts?

**Comment by jayeshmangwani** [comment]
> Is it enabled by default for newly created accounts?

For me, that's not the case. I created a new user in NewDot but didn't receive any notifications from the other user. However, I do get notifications when I trigger the console part in OldDot.

**Comment by perunt** [comment]
I was able to get notifications working. I'm going to push a temporary PR with the fix so we don't have the issue in production. Then we'll fix it on the native side. Does this work for you?

**Comment by jayeshmangwani** [comment]
Yes, that sounds good to me as long as the temporary fix doesn't cause any regressions. @puneetlath, are you okay with the plan? We'll add the temp fix for now and handle the issue on the native side afterward.

**Comment by wildan-m** [comment]
I don't know if temporary fix allowed, usually upstream fix enforced or that temporary will ended up become permanent.

**Comment by perunt** [comment]
`Temporary fix` sounds kinda hacky here. FYI, we shouldn't ditch this fix later since this bug also appears due to some mess-ups in the theme handling PR. So basically, we're gonna stick with this fix, and the other stuff will get added upstream.

**Comment by puneetlath** [comment]
Ok let's do it. Can you raise a PR?

**Comment by wildan-m** [comment]
The temporary fix looks familiar, unfortunately common contributor isn't allowed to apply temporary fix. I believe the C and C+ shouldn't be paid until upstream PR deployed to prod (?)

**Comment by jayeshmangwani** [comment]
I don't think we need a dedicated regression test for this bug. It's an obvious UI issue that should be caught immediately if it breaks again. @puneetlath , WDYT ?

**Comment by puneetlath** [comment]
That makes sense to me.

**Comment by puneetlath** [comment]
Payment summary: 
- $250 - @jayeshmangwani - C+ - to be paid via NewDot

**Comment by wildan-m** [comment]
@puneetlath why this is closed and paid? where is the upstream PR? 

The merged PR is essentially the same with my proposed solution. How compensation works in this case?

discussing in slack if no satisfactory answer within 24 hours.

**Comment by wildan-m** [comment]
@shubham1206agra every issues has help wanted tag removed after assignemnt. Please check carefully, I've send the proposal after help wanted tag added.

![Screenshot_20241010-081727](https://github.com/user-attachments/assets/60ba772d-8e7b-48bc-8f0c-b2f27aaa940b)

**Comment by wildan-m** [comment]
@shubham1206agra can you send me the links to similar cases? where that 1/5 partial payment come from?

**Comment by jayeshmangwani** [comment]
> discussing in slack if no satisfactory answer within 24 hours.

You can start a discussion if you'd like to.

**Comment by jayeshmangwani** [comment]
> The merged PR is essentially the same with my proposed solution. How compensation works in this case?

Yes, the merged PR uses the same logic as your proposal. Regarding the compensation, I'm not sure if it will be a partial or full payout, so I'll leave that decision to @francoisl and @puneetlath.

**Comment by puneetlath** [comment]
@perunt just checking in. Mind providing an update here?

**Comment by jayeshmangwani** [comment]
Hey @perunt , do we have any plans for an upstream fix? Can you please provide an [update here](https://github.com/Expensify/App/issues/48190#issuecomment-2405391028)?

**Comment by jayeshmangwani** [comment]
Waiting on an update from Taras. bumped on [Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1729676708143789)

**Comment by puneetlath** [comment]
Bumped in Slack again.

**Comment by perunt** [comment]
hey guys, quick update. I got a bit sidetracked with integrating the Nitro Module into the app, but I’m aiming to wrap it up by the end of this week and get back to this task. As I mentioned before, some changes were made to one of our libraries that should resolve the issue, and I'll test it once I finish the Contact Module

**Comment by puneetlath** [comment]
@perunt any update here?

**Comment by perunt** [comment]
I just tested it again, and it's reproducible. 
I dug deeper and found this section: [MainActivity.kt, line 50](https://github.com/Expensify/App/blob/main/android/app/src/main/java/com/expensify/chat/MainActivity.kt#L50). Here, we're setting the top padding to zero to avoid jumpy behavior with the splash screen. 
Should we spend more time on this and find a consensus between addressing the bug and handling it for the splash screen, or can we stick with replacing the status bar height using the value from `react-native-safe-area-context` (our current fix that was merged)?

**Comment by puneetlath** [comment]
I think let's just go with what's easiest. I don't think it's worth spending a lot of time on right now.

**Comment by jayeshmangwani** [comment]
Just like Shubham, I am also able to reproduce this issue again in the latest Play Store app.

**Comment by perunt** [comment]
Can you drop details here? @shubham1206agra @jayeshmangwani

**Comment by jayeshmangwani** [comment]
@perunt If you’re asking about the steps I followed to reproduce this issue, here they are:

1. Logged in as User A on an Android device (app is closed).
2. Sent a message from User B to User A.
3. User A received a notification and tapped on it.
4. Upon navigation to the Report Screen, the top area is missing.

**Comment by jayeshmangwani** [comment]
@perunt @puneetlath we can still easily reproduce this issue on the latest version, so it has reappeared.

What are the next steps to fix this issue?

**Comment by jayeshmangwani** [comment]
@perunt It seems that our changes have been reverted in this PR https://github.com/Expensify/App/pull/53223. Currently, we are not passing `safeAreaInsets.top` as the fallback value for top in `src/styles/utils/getSafeAreaInsets/index.android.ts`
https://github.com/Expensify/App/blob/9ac63569b4dcdd1ed54fab4b7d63bcf1122835fa/src/components/SafeAreaConsumer/index.tsx#L16-L18
https://github.com/Expensify/App/blob/3cf603787a6654096e5acf5139ff83e6a672fca0/src/styles/utils/getSafeAreaInsets/index.android.ts#L8-L12

**Comment by perunt** [comment]
@jayeshmangwani  Thanks for the investigation!

I asked Chris to double-check the padding issue some time ago. Do you @chrispader have any updated information on this? I'm mainly interested in whether it was a mistake and you simply cleaned the code, or if you encountered another bug.

**Comment by jayeshmangwani** [comment]
Is anyone able to reproduce this issue now? I am currently unable to. @shubham1206agra, since you initially raised the issue, are you able to reproduce it now?

**Comment by jayeshmangwani** [comment]
@puneetlath I have tested this issue a few times on two different devices and can’t reproduce it now. @puneetlath @perunt, what should we do here? Should we close this or keep it open to investigate further for the [upstream PR](https://github.com/Expensify/App/issues/48190#issuecomment-2405391028)?

**Comment by chrispader** [comment]
@jayeshmangwani i'll check today if https://github.com/Expensify/App/pull/57128 potentially fixes this, not sure though

**Comment by chrispader** [comment]
I can also not reproduce this issue at the moment. Tested on two Samsung devices, one with notch and one without.

**Comment by puneetlath** [comment]
Ok if no one can reproduce, I'm going to close it. Comment if you disagree or if we missed paying anyone out or anything.
